pkg/syncer/reconcile/fight_detector.go

```diff
@@ -57,12 +58,15 @@ func FightWarning(frequency float64, resource client.Object) status.ResourceErro
 // Instantiate with newFightDetector().
 //
 // Performance characteristics:
-//  1. Current implementation is NOT threadsafe.
+//  1. Current implementation is threadsafe.
 //  2. Current implementation has unbounded memory usage on the order of the
 //     number of objects the Syncer updates through its lifetime.
 //  3. Updating an already-tracked resource requires no memory allocations and
 //     take approximately 30ns, ignoring logging time.
 type fightDetector struct {
```

Comment:
I think perhaps this should be defined in it's own package with public methods to safeguard against future changes directly accessing the protected variables. Currently it looks like this is all used within the same package, so the private fields aren't really protected semantically.

Reply:
That'a s great point. Done.